### ISSUE TITLE
default_workspace isn't obeyed by mux servers

### ISSUE THREAD

**Issue by Lenbok** [wez/wezterm#2981]
### What Operating System(s) are you seeing this problem on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

xfce

### WezTerm version

wezterm 20230115-214820-a2e027df

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

I don't want any workspace named "default", so have set the `default_workspace` argument in my lua config.

When I start wezterm, the default workspace is now whatever I said in my config.
If I then create a workspace on the current domain, suddenly a workspace named "default" appears.

### To Reproduce

Start wezterm with the provided config (hacked down from my main config, and uses backtick as leader).
Note (via backtick-s) that there is only one workspace (named after the current host).
Create a workspace (via backtick-shift-s or backtick-shift-p).
Note (via backtick-s) that there is now an extra "default" workspace.

### Configuration

```lua
local wezterm = require("wezterm");

local hostname = wezterm.hostname();

-- Dynamically build the unix domains configuration element.
-- This lets us use an ssh proxy for hosts except for the current host.
--
-- Note: We are using an ssh proxy to permit port forwarding (since wezterm
-- native ssh domains do not permit this.)
local unix_hosts = {"localhost", "ruru", "noir", "tron" }
local unix_domains = {}
while #unix_hosts > 0 do
   local host = table.remove(unix_hosts, 1)
   if host == "localhost" or host == hostname then
      table.insert(unix_domains, {name = host  })
   else
      table.insert(unix_domains, {name = host, proxy_command = {"ssh", "-T", host, "wezterm", "cli", "proxy"}})
   end
end

return {
   default_workspace = hostname,
   unix_domains = unix_domains,
   disable_default_key_bindings = true,
   exit_behavior = "Close",

   enable_tab_bar = true,  -- false when using tmux, true for wezterm multiplexing

   leader = { key = "`", timeout_milliseconds = 2000}, -- use this when you're not using tmux. Add mods = "CTRL" to disable if using tmux
   keys = {
      -- Send "`" key to the terminal when pressed after LEADER
      { key = "`", mods = "LEADER",       action = wezterm.action{SendString = "`"}},
      { key = "\t",mods = "LEADER",       action = "ActivateLastTab"},
      { key = " ", mods = "LEADER",       action = "ShowLauncher"},
      { key = " ", mods = "LEADER|CTRL",       action = wezterm.action{ShowLauncherArgs = {flags = "FUZZY|DOMAINS|WORKSPACES|TABS|LAUNCH_MENU_ITEMS|KEY_ASSIGNMENTS"}}},

      { key = "s", mods = "LEADER",       action = wezterm.action{ShowLauncherArgs = {title = "Workspaces", flags = "FUZZY|WORKSPACES"}}},
      -- Experimental, associate workspace with domain, see https://github.com/wez/wezterm/issues/1874
      -- Should switch to the "default" workspace, (named after this host).
      { key = "l", mods = "LEADER|CTRL",       action = wezterm.action{SwitchToWorkspace = {
                                                                     name = hostname,
                                                                     spawn = {
                                                                        domain = {
                                                                           DomainName = hostname
                                                                        }
                                                                     }
      }}},
      -- A project workspace, on the current host
      { key = "s", mods = "LEADER|SHIFT",       action = wezterm.action{SwitchToWorkspace = {
                                                                     name = "SomethingElse",
                                                                     spawn = {
                                                                        domain = {
                                                                           DomainName = hostname
                                                                        }
                                                                     }
      }}},
      -- A project workspace, on a remote host
      { key = "p", mods = "LEADER|SHIFT",       action = wezterm.action{SwitchToWorkspace = {
                                                                     name = "Project X",
                                                                     spawn = {
                                                                        domain = {
                                                                           DomainName = "ruru"
                                                                        }
                                                                     }
      }}},


   },
}

```

### Expected Behavior

A workspace named "default" should not magic itself into existence.

**Comment by wez** [comment]
Do you have `default_workspace` set in the config on the mux server?

**Comment by wez** [comment]
Would be interesting to see the output of `wezterm cli list` on the mux server, as well as locally.

**Comment by Lenbok** [comment]
Yes, since I can reproduce it when the domain is the current host, their configs are the same (and I killed any existing mux server before starting this)

In terms of attempting `cli list`, it doesn't work:
```
$ ./Downloads/WezTerm-nightly-Ubuntu18.04.AppImage --config-file wezterm.lua &
[1] 2147206

$ ./Downloads/WezTerm-nightly-Ubuntu18.04.AppImage --config-file wezterm.lua cli list
16:14:29.855  ERROR  wezterm > failed to connect to Socket("/run/user/1672/wezterm/gui-sock-2083333"): connecting to /run/user/1672/wezterm/gui-sock-2083333; terminating

[unsurprising as at this point I hadn't created a new workspace. I then used backtick-P to create a new workspace, which triggered this on stdout]

16:14:36.659  WARN   wezterm_client::client > While connecting to Socket("/run/user/1672/wezterm/sock"): connecting to /run/user/1672/wezterm/sock.  Will try spawning the server.
16:14:36.659  WARN   wezterm_client::client > Running: "/tmp/.mount_WezTer3aNrSI/usr/bin/wezterm-mux-server" "--daemonize"

[now try the list again, which also fails:]

$ ./Downloads/WezTerm-nightly-Ubuntu18.04.AppImage --config-file wezterm.lua cli list
16:15:00.295  ERROR  wezterm > failed to connect to Socket("/run/user/1672/wezterm/gui-sock-2083333"): connecting to /run/user/1672/wezterm/gui-sock-2083333; terminating

[at this point, backtick-s shows there is a "default" workspace in existence]
```

**Comment by wez** [comment]
Speculatively fixed in `main`; could you give it a try once binaries are available?

**Comment by Lenbok** [comment]
Tried with the new build and I think this particular issue is fixed. Something is buggy with the window handling with respect to workspaces though, but I will file a separate issue for that.
